hardware/megacli.py

```diff
@@ -195,31 +198,93 @@ def ld_get_num(ctrl):
 
 
 def ld_get_info(ctrl, ldrv):
-    'Get info about a logical drive on a controller.'
+    """Get info about a logical drive on a controller."""
     return run_and_parse('LDInfo -L%d -a%d' % (ldrv, ctrl))
 
 
-if __name__ == "__main__":
-    import pprint
-
-    for ctrl_num in range(adp_count()):
-        print('Controler', ctrl_num)
-        pprint.pprint(adp_all_info(ctrl_num))
-
-        encs = enc_info(ctrl_num)
-
-        print()
-
-        print('Enclosing:')
-        pprint.pprint(encs)
-
-        for enc in encs:
-            for disk_num in range(enc['NumberOfPhysicalDrives']):
-                print()
-                print('Physical disk', disk_num)
-                pprint.pprint(pdinfo(ctrl_num, enc['DeviceId'], disk_num))
-
-        for ld_num in range(ld_get_num(ctrl_num)):
-            print()
-            print('Logical disk', ld_num)
-            pprint.pprint(ld_get_info(ctrl_num, ld_num))
+def detect():
+    """Detect LSI MegaRAID controller configuration."""
+    hw_lst = []
+    ctrl_num = adp_count()
+    disk_count = 0
+    global_pdisk_size = 0
+    if ctrl_num > 0:
```

Comment:
I would invert the logic here as all the rest of the code is indented again this.
Retuning an explicit value for  ctrl_num==0 and then continue the rest of the logic with the final return

Reply:
yeah, good call, I'll change it